### PR TITLE
Registration Page FE

### PR DESCRIPTION
**Pull request by BlasAlbert**
Task Link: https: https://app.asana.com/0/1202718495875213/1202725278795526/f

Definition of Done

Follow UI design
Applied Modal

Commands

cd frontend
npm install
npm start

Screenshot
image

### CODE REVIEW

File: frontend/src/Navbar/NavBar.css [hunk 1]
```diff
@@ -1,3 +1,27 @@
...
+  }
+
+  .signup-btn:hover {
+	cursor: pointer;
+	background: rgb(212, 34, 34);
+	transition: all 0.4s ease-out;
+  }
+
+
+
```
Comment (chevyB) [inline frontend/src/Navbar/NavBar.css]: Just add only 1 line at the end of file

Comment (chevyB) [inline frontend/src/Navbar/NavBar.css:27]: Add only 1 new line at the end of file

File: frontend/src/Navbar/NavBar.js [hunk 2]
```diff
@@ -1,8 +1,18 @@
...
 import { useNavigate } from 'react-router-dom';
 import './NavBar.css';
+import { Button, Modal } from 'semantic-ui-react'
+import FormSignup from '../RegistrationPage/FormSignup';
+import FormSuccess from '../RegistrationPage/FormSuccess';
+
 const NavBar = () => {
 	const navigate = useNavigate();
+	const [isSubmitted, setIsSubmitted] = useState(false);
+	const [open, setOpen] = React.useState(false)
```
Comment (chevyB) [inline frontend/src/Navbar/NavBar.js]:
```suggestion
	const [open, setOpen] = useState(false)
```

File: frontend/src/RegistrationPage/Form.css [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+  .videoModal .modal-body{
+      padding: 9px 10px;
+  }
+  .videoModal .modal-header .close {
+      color: #151515;
+  }
+  .modal-open .header {
+      padding-right: 17px;
+  }*/
+
```
Comment (chevyB) [inline frontend/src/RegistrationPage/Form.css]: Are these needed? If not, pls remove

File: frontend/src/RegistrationPage/Form.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+          <FormSuccess />
+        )}
+      </div>
+
+    </>
+  );
+};
+
+
+
```
Comment (chevyB) [inline frontend/src/RegistrationPage/Form.js]:
```suggestion
```
Remove extra white space

File: frontend/src/RegistrationPage/Form.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+export default Registration_Page;
```
Comment (chevyB) [inline frontend/src/RegistrationPage/Form.js]: add new line always at the end of file

File: frontend/src/RegistrationPage/FormSignup.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        </button>
+        <span className='form-input-login'>
+          Already have an account? <a href='#'>Login</a>
+        </span>
+      </form>
+    </div>
+  );
+};
+
+export default FormSignup;
```
Comment (chevyB) [inline frontend/src/RegistrationPage/FormSignup.js]: add new line at EOF

File: frontend/src/RegistrationPage/FormSuccess.js [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+
+const FormSuccess = () => {
+  return (
+    <div className='form-content-right'>
+      <img className='form-img-2' src='Images/Check.png' alt='success' />
+    </div>
+  );
+};
+
+export default FormSuccess;
```
Comment (chevyB) [inline frontend/src/RegistrationPage/FormSuccess.js]: add new line

File: frontend/src/RegistrationPage/useForm.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+        callback();
+      }
+    },
+    [errors]
+  );
+
+  return { handleChange, handleSubmit, values, errors };
+};
+
+export default useForm;
```
Comment (chevyB) [inline frontend/src/RegistrationPage/useForm.js]: add new line

File: frontend/src/RegistrationPage/validateInfo.js [hunk 9]
```diff
@@ -0,0 +1,28 @@
+export default function validateInfo(values) {
+    let errors = {};
+  
+    if (!values.username.trim()) {
+      errors.username = 'Username required';
+    }
+    // else if (!/^[A-Za-z]+/.test(values.name.trim())) {
+    //   errors.name = 'Enter a valid name';
+    // }
```
Comment (chevyB) [inline frontend/src/RegistrationPage/validateInfo.js]: remove if not needed anymore

File: frontend/src/RegistrationPage/validateInfo.js [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+      errors.password = 'Password needs to be 6 characters or more';
+    }
+  
+    if (!values.password2) {
+      errors.password2 = 'Password is required';
+    } else if (values.password2 !== values.password) {
+      errors.password2 = 'Passwords do not match';
+    }
+    return errors;
+  }
```
Comment (chevyB) [inline frontend/src/RegistrationPage/validateInfo.js]: add new line

**Comment by BlasAlbert** [comment]
Committed and pushed  the changes sir...

**Review by chevyB — APPROVED** [review]

**Pull request merged**